### PR TITLE
Adding files for Lesson 09

### PR DESCRIPTION
**Pull request by mattcasali**
Added file for Lesson 09. Please review. Thanks!

### CODE REVIEW

File: students/matt_casali/Lesson09/mailroom_9.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+def get_amount():
+    try:
+        return float(input(f"How much did this person donate? "))
+    except ValueError:
+        print("You did not enter a numeric value. Please try again.")
+        sys.exit()
+
+
+class Donor_Actions:
```
Comment (natasha-aleksandrova) [inline students/matt_casali/Lesson09/mailroom_9.py]: per instructions your data classes should not contain user interactions (calls to `input()`)

your second data class should deal with a collection of Donor objects.

File: students/matt_casali/Lesson09/mailroom_9.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    @property
+    def full_name(self):
+        return f"{self.first} {self.last}"
+
+    def add_donation(self, amount):
+        return self.donation.append(amount)
+
+    def sum_donation(self):
+        return sum(self.donation)
```
Comment (natasha-aleksandrova) [inline students/matt_casali/Lesson09/mailroom_9.py]: you can also include methods for average, number of gifts and letter text.

File: students/matt_casali/Lesson09/mailroom_9.py (hunk 1 above)
Comment (mattcasali) [reply students/matt_casali/Lesson09/mailroom_9.py]: Hey Natasha, I made sure to keep the functions involving and calls to input() outside of the two classes I created. Those two functions are called on within the Donor_Actions class. Is there a better way I should approach this problem? Thanks!

Comment (natasha-aleksandrova) [reply students/matt_casali/Lesson09/mailroom_9.py]: Even though `input()` isn't called directly - it calls a function that wraps the input function, which ends up being the same.


Your two data classes should be purely dealing with data then your program flow code will interact with these objects.

Comment (mattcasali) [reply students/matt_casali/Lesson09/mailroom_9.py]: Okay, I think that makes sense. I've added the calls for input() to the main() function so the calls stay outside of the Donor_Actions class. 

I also went back and added more methods for average donation, the number of donations, the text within the thank you letter, and a few others. 

Would it be possible for to resubmit the exercise?

Comment (natasha-aleksandrova) [reply students/matt_casali/Lesson09/mailroom_9.py]: Absolutely - resubmit via a new PR and I will review/adjust the grade accordingly.

Comment (mattcasali) [reply students/matt_casali/Lesson09/mailroom_9.py]: Hey Natasha, just submitted another PR with the new code.

https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1493

Thanks for the help with this!

**Pull request merged**